File: app-mvvm/src/main/java/com/xuie/imaginary/gui/gank/meizhi/MeiZhiAdapter.java
package com.xuie.imaginary.gui.gank.meizhi;

import com.xuie.imaginary.R;
import com.xuie.imaginary.base.BaseDataBindingAdapter;
import com.xuie.imaginary.data.BaseBean;
import com.xuie.imaginary.databinding.ItemFragmentMeizhiBinding;

import java.util.List;

/**
 * @author xuie
 * @date 17-7-5
 */
public class MeiZhiAdapter extends BaseDataBindingAdapter<BaseBean, ItemFragmentMeizhiBinding> {
    MeiZhiAdapter(List<BaseBean> meiZhis) {
        super(R.layout.item_fragment_meizhi, meiZhis);
    }

    @Override
    protected void convert(ItemFragmentMeizhiBinding binding, BaseBean item) {
        binding.setData(item);
    }
}

File: app-mvvm-kodein/src/main/kotlin/xuk/imaginary/gui/gank/meizhi/MeiZhiAdapter.kt
package xuk.imaginary.gui.gank.meizhi

import xuk.imaginary.R
import xuk.imaginary.base.BaseDataBindingAdapter
import xuk.imaginary.data.BaseBean

/**
 * @author Jie Xu
 * @date 17-7-5
 */
class MeiZhiAdapter internal constructor(meiZhis: List<BaseBean>) :
    BaseDataBindingAdapter<BaseBean, xuk.imaginary.databinding.ItemFragmentMeizhiBinding>(R.layout.item_fragment_meizhi, meiZhis) {

  override fun convert(binding: ItemFragmentMeizhiBinding?, item: BaseBean) {
    binding!!.data = item
  }
}
